### PR TITLE
Move events to submodules

### PR DESCRIPTION
**Pull request by sug0**
## Describe your changes

Move `Event` and associated types from `namada_core` to the new `namada_events` crate. Module-specific events (such as `IbcEvent` and `EthBridgeEvent`) were moved to their respective crates (`namada_ibc` and `namada_ethereum_bridge`).

~~**NOTE:** IBC e2e tests are expected to fail, just like in the base PR.~~

## Indicate on which release or other PRs this topic is based on

#3088

Diff for review: https://github.com/anoma/namada/compare/tiago/emit-generic-events-wasm..tiago/move-events-to-submodules (https://github.com/anoma/namada/pull/3102/files/0ac23e8a2d514d30a94abb42eafc8c0d6f5ccebd..44d0cdbce373910043013bc75da8cd91a4408cee)

## Checklist before merging to `draft`
- [x] I have added a changelog
- [x] Git history is in acceptable state

### CODE REVIEW

**Comment by sug0** [comment]
ci is temporarily borked due to `sccache`

File: crates/governance/src/event.rs [hunk 1]
```diff
@@ -0,0 +1,265 @@
...
+                    proposal_id,
+                    false,
+                    result,
+                ),
+            ),
+        };
+
+        let mut event = Self::new(event_type, EventLevel::Block);
+
+        #[allow(deprecated)]
```
Comment (tzemanovic) [inline crates/governance/src/event.rs:181]: should we open an issue for replacing these?

**Review by tzemanovic — DISMISSED** [review]

File: crates/governance/src/event.rs (hunk 1 above)
Comment (sug0) [reply crates/governance/src/event.rs:181]: yeah, ideally

File: crates/ibc/src/event.rs [hunk 2]
```diff
@@ -60,6 +61,28 @@ pub mod types {
...
+    const DOMAIN: &'static str = "ibc";
+}
+
+impl From<IbcEvent> for Event {
+    fn from(ibc_event: IbcEvent) -> Self {
+        let mut event = Self::new(
+            EventTypeBuilder::new_of::<IbcEvent>()
+                .with_segment(ibc_event.event_type.0)
+                .build(),
+            EventLevel::Tx,
```
Comment (grarco) [inline crates/ibc/src/event.rs:74]: Do we emit the ibc events at the tx or block level?

**Review by grarco — APPROVED** [review]
Just one comment but looks good to me!

File: crates/ibc/src/event.rs (hunk 2 above)
Comment (sug0) [reply crates/ibc/src/event.rs:74]: most are tx level events. pgf over ibc is at the block level, but idk a clean solution to change the event level there lol

File: crates/ibc/src/event.rs [hunk 3]
```diff
@@ -0,0 +1,521 @@
...
+fn validate_ibc_event_type(
+    namada_event: &Event,
+) -> Result<IbcEventType, EventError> {
+    if namada_event.kind().domain() != IbcEvent::DOMAIN {
+        return Err(EventError::InvalidEventType);
+    }
+
+    let event_type = namada_event.kind().sub_domain();
+
+    // if !matches!(
```
Comment (batconjurer) [inline crates/ibc/src/event.rs:149]: what's going on here?

Comment (sug0) [reply crates/ibc/src/event.rs:149]: I'm not sure if you're referring to the let expr, or the commented `matches!()`. the let expr is to strip the `ibc/` prefix from the event type, which `IbcEvent` instances don't contain. the `matches!()` was to check if the sub-domain was valid, but I'm not sure which ones are currently in use in namada...

**Review by batconjurer — DISMISSED** [review]

**Pull request merged**